google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -761,13 +773,21 @@ private static WriteLogEntriesRequest writeLogEntriesRequest(
   }
 
   private static LogName getLogName(
-      String projectId, String logName, LogDestinationName destination) {
-    if (logName == null) {
+      String projectId,
+      String logName,
+      LogDestinationName destination,
+      boolean validateParameters) {
+    if (validateParameters) {
+      Preconditions.checkNotNull(logName, "logName parameter cannot be null");
+    } else if (logName == null) {
```

Comment:
it is not best solution. more visual and easy to understand would be to validate for NPE before you call this (private) method and keep previous implementation or _always_ throw NPE if required argument(s) are `null`.

Reply:
I agree that in this case we can move parameters validation into `deleteLogAsync` - we can always move those validation into `getLogName()` in case if it will be called from more places in the future (since otherwise we are duplicating checks all over the code).